From d586077a37e14446cf385893608ea9e60a59d083 Mon Sep 17 00:00:00 2001
From: Brad Smith <brad@comstyle.com>
Date: Sun, 23 Aug 2026 01:26:50 -0400
Subject: [PATCH] Replace deprecated OpenSSL API ASN1_STRING_data()

ASN1_STRING_data() has been deprecated since OpenSSL 1.1.0. Replace
with ASN1_STRING_get0_data(). ASN1_STRING_data() has been removed
with OpenSSL 4.

Upstream: https://github.com/apache/thrift/commit/d586077a37e14446cf385893608ea9e60a59d083

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 lib/cpp/src/thrift/transport/TSSLSocket.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/cpp/src/thrift/transport/TSSLSocket.cpp b/lib/cpp/src/thrift/transport/TSSLSocket.cpp
index bb8dfd13f..48b96ebf7 100644
--- a/lib/cpp/src/thrift/transport/TSSLSocket.cpp
+++ b/lib/cpp/src/thrift/transport/TSSLSocket.cpp
@@ -770,7 +770,11 @@ void TSSLSocket::authorize() {
       if (name == nullptr) {
         continue;
       }
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+      const char* data = (const char*)ASN1_STRING_get0_data(name->d.ia5);
+#else
       char* data = (char*)ASN1_STRING_data(name->d.ia5);
+#endif
       int length = ASN1_STRING_length(name->d.ia5);
       switch (name->type) {
       case GEN_DNS:
-- 
2.47.3

